caravel/assets/visualizations/mapbox.jsx

```diff
@@ -0,0 +1,336 @@
+            gradient.addColorStop(1, "rgba(" + rgb[1] + ", " + rgb[2] + ", " + rgb[3] + ", 0.8)");
+            gradient.addColorStop(0, "rgba(" + rgb[1] + ", " + rgb[2] + ", " + rgb[3] + ", 0)");
+            ctx.fillStyle = gradient;
+            ctx.fill();
+
+            if (this._isNumeric(clusterLabel)) {
+              clusterLabel = clusterLabel >= 10000 ? Math.round(clusterLabel / 1000) + 'k' :
+                             clusterLabel >= 1000 ? (Math.round(clusterLabel / 100) / 10) + 'k' : clusterLabel;
+              this._drawText(ctx, pixelRounded, fontHeight, clusterLabel, scaledRadius, rgb, true);
+            }
+          } else {
+            var defaultRadius = radius / 6,
+                radiusProperty = location.get("properties").get("radius"),
+                pointRadius = radiusProperty === null ? defaultRadius : radiusProperty,
+                pointLabel,
+                pointMetric = location.get("properties").get("metric");
+
+             if (radiusProperty !== null) {
+              if (props.pointRadiusUnit === "Kilometers") {
+                pointLabel = d3.round(pointRadius, 2) + "km";
+                pointRadius = this._kmToPixels(pointRadius, props.latitude, props.zoom);
+              } else if (props.pointRadiusUnit === "Miles") {
+                pointLabel = d3.round(pointRadius, 2) + "mi";
+                pointRadius = this._kmToPixels(pointRadius * milesToKm, props.latitude, props.zoom);
+              }
+            }
+
+            if (pointMetric !== null) {
+              pointLabel = this._isNumeric(pointMetric) ? d3.round(pointMetric, 2) : pointMetric;
+            }
+
+            // Fall back to default points if pointRadius wasn't a numerical column
+            if (!pointRadius) {
+              pointRadius = defaultRadius;
+            }
+
+            ctx.arc(pixelRounded[0], pixelRounded[1], d3.round(pointRadius, 1), 0, Math.PI * 2);
+            ctx.fillStyle = "rgb(" + rgb[1] + ", " + rgb[2] + ", " + rgb[3] + ")";
+            ctx.fill();
+
+            if (pointLabel !== undefined) {
+              var fontHeight = d3.round(pointRadius, 1);
+              this._drawText(ctx, pixelRounded, fontHeight, pointLabel, pointRadius, rgb, false);
+            }
+          }
+        }
+      }, this);
+    }
+
+    ctx.restore();
+  }
+}
+
+class MapboxViz extends React.Component {
+  constructor(props) {
+    super(props);
+
+    var longitude = this.props.viewportLongitude || -122.405293,
+        latitude = this.props.viewportLatitude || 37.772123;
+
+    this.state = {
+      viewport: {
+        longitude: longitude,
+        latitude: latitude,
+        zoom: this.props.viewportZoom || 11,
+        startDragLngLat: [longitude, latitude]
+      }
+    };
+
+    this.onChangeViewport = this.onChangeViewport.bind(this);
+  }
+
+  onChangeViewport(viewport) {
+    this.setState({
+      viewport: viewport
+    });
+  }
+
+  render() {
```

Comment:
I'm leaning towards not destructing since there are would be too many things to unwrap (with only a handful being used more than once)